lib/web/apiserver.go

```diff
@@ -350,6 +350,8 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 	h.POST("/webapi/mfa/login/finish", httplib.MakeHandler(h.mfaLoginFinish))
 	h.POST("/webapi/mfa/login/finishsession", httplib.MakeHandler(h.mfaLoginFinishSession))
 	h.GET("/webapi/mfa", h.WithAuth(h.getMFADevicesHandle))
+	h.GET("/webapi/mfa/:token", httplib.MakeHandler(h.getMFADevicesWithTokenHandle))
```

Comment:
A few remarks here:

1) `GET /webapi/mfa` doesn't make much conceptual sense - we are not fetching MFAs, we are fetching devices. I suggest "GET /webapi/mfa/devices" (ie, having a "devices" collection).

2) If we have `GET /webapi/mfa/devices/:something`, then I'd expect `:something` to be a device name or ID, not the token. Let's push the token to a parameter (preferred) or to the request body.

3) Per suggestion above, let's unify the endpoints. If the token is present you use it, otherwise it's the context user. API surface is big enough already.

```suggestion
	h.GET("/webapi/mfa/devices", httplib.MakeHandler(h.getMFADevices))
```

Reply:
revised to use query param

Reply:
@kimlisa  please add `makeMFAHandler` middleware wrapper,  this will make the security aspect of these endpoints more declarative as `httplib.MakeHandler` is used for public endpoints.

Reply:
I actually can't do a wrapper here b/c the public/private handler has different signatures. 

It we want to be declarative, i think it makes more sense to separate it into specific public and private endpoints

```
  GET /mfa/:token/devices  // public endpoint, requires a token
  GET /mfa/devices

  // Similar public signature for deleting a device
  DELETE /mfa/:token/devices/:deviceName
```

Reply:
If you want to split, I'd push the token to the end of the URL - I don't think `:token` makes sense under `/mfa`.

This looks like some other endpoints we have (ie, `/users/password/token/`):

```
GET /mfa/devices/token/:token                     
    # not great IMO, sounds like GET-mfa-device-token, which implies that a
    # "device token" exists
DELETE GET /mfa/devices/:deviceName/token/:token
    # as above, not the best, sounds like we are deleting a "device token"
```

or maybe:

```
GET /mfa/devicesByToken/:token
    # sounds like GET-mfa-device-by-token, seems appropriate
DELETE /mfa/deviceByToken/:token
    # sounds like DELETE-mfa-device-by-token, also appropriate
    # name pushed to params/body, which is a bit weird
```

Another possibility would be to use [custom methods](https://cloud.google.com/apis/design/custom_methods) - those have a literal `:` in the URL, but idk how that plays out with our routing APIs. It would look like this:

```
GET /mfa/devices:byToken                  # literal `:`, token passed by param or body
DELETE /mfa/devices/<deviceName>:byToken  # literal `:`, token passed by param or body
```

Finally, we could take the idea above and replace the literal `:` with a slash, giving:

```
GET /mfa/devices/byToken                  # token passed by param or body
DELETE /mfa/devices/<deviceName>/byToken  # token passed by param or body
```

Reply:
The comment above was a bit of a stream of consciousness, but I think it has value so you can follow what I'm thinking about in terms of designing REST URLs. I would take any of the options above, but I'd favor the last 2, as I think they 1) have better clarity and 2) we get the deviceName as the ID in the URL (with a preference for the token as a parameter rather than in the body - neither GET or DELETE typically have a body).

Reply:
revised with the last suggestion

Reply:
alexey pointed out potential problems, one of which is, if we ever decide to just retrieve one device for whatever reason this endpoint `GET /mfa/devices/byToken ` will  conflict with this endpoint `GET /mfa/devices/:devicename`

This was alexey's suggestion: 
```
/webapi/mfa/token/:tokenID/devices
/webapi/mfa/token/:token/devices/:devicename
```

I actually prefer alexey's suggestion versus adding the `token/:tokenid` at the end of the URL, b/c to me it says "authn with token, then get devices". It also lets reader know "to use this endpoint, i need a token for etc". I also think its simpler to keep consistent when we get more of these type of endpoints.